src/core/videosource.cpp

```diff
@@ -201,6 +201,9 @@ FFMS_VideoSource::FFMS_VideoSource(const char *SourceFile, FFMS_Index &Index, in
         if (CodecContext->codec_id == AV_CODEC_ID_VC1 && CodecContext->has_b_frames)
             CodecContext->has_b_frames = 7; // the maximum possible value for vc1
```

Comment:
You should probably assign Delay here instead of CodecContext->has_b_frames since you made the effort to turn it into a separate variable. VC1 has a secret internal value unlike h264 and doesn't care if it's set.